packages/block-editor/src/components/block-list/block.js

```diff
@@ -290,7 +290,7 @@ function BlockListBlock( {
 	 * (via `setFocus`), typically if there is no focusable input in the block.
 	 */
 	const onFocus = () => {
-		if ( ! isSelected && ! isAncestorOfSelectedBlock && ! isPartOfMultiSelection ) {
+		if ( ! isSelected && ! isPartOfMultiSelection ) {
```

Comment:
I know this is a crucial part to fixing the bug 😅 , but this change has caused the original nav item appender bug to resurface in Firefox (though not in Chrome):
![ezgif com-optimize](https://user-images.githubusercontent.com/8096000/70881344-4c567c00-2020-11ea-8404-1e4458a543d8.gif)
cc @draganescu

Reply:
Yes, Firefox is the issue here and why the whole original attempt was so convoluted :D

Reply:
Hm, reading this and #18379, I might expect this could be related a Firefox+macOS specific treatment of click events in buttons in which the clicks do not cause focus:

https://gist.github.com/cvrebert/68659d0333a578d75372
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus

Reply:
There's a few related handlings of this in Gutenberg elsewhere:

https://github.com/WordPress/gutenberg/blob/74269e6956392f66622af458b1bb7e8ab1dac3fb/packages/block-editor/src/components/block-list/insertion-point.js#L54-L60

https://github.com/WordPress/gutenberg/blob/74269e6956392f66622af458b1bb7e8ab1dac3fb/packages/block-editor/src/components/default-block-appender/index.js#L38-L49

https://github.com/WordPress/gutenberg/blob/74269e6956392f66622af458b1bb7e8ab1dac3fb/packages/components/src/higher-order/with-focus-outside/index.js#L100-L121

Similarly, I'm thinking it could be enough to add a `tabIndex={ -1 }` to the `div` wrapper, which could allow the focus event to occur when the appender is clicked, while not adding an extra tab stop.

Reply:
> Similarly, I'm thinking it could be enough to add a `tabIndex={ -1 }` to the `div` wrapper, which could allow the focus event to occur when the appender is clicked, while not adding an extra tab stop.

This seems to work in my testing. See 9091e9701, which includes the test and an explanatory inline comment.

Reply:
It's working fine for me now 🎉